### PR TITLE
feat(ODD): HiFi designs and features for the protocol details page

### PR DESCRIPTION
**Pull request by ewagoner**
# Overview

This PR implements the designs and features found in Figma at https://www.figma.com/file/AoTLAYuWawlaWItB1umOjr/Opentrons-Flex-Touchscreen-Designs?node-id=1-212&t=Y7Cv06vknVeA2a4h-0

TODO Items:
- Set the chip contents at the top of the page
- Use a snackbar component to show pin/unpin result
- Add liquid sub-page
- Use final design for deck map sub-page
- Use iconography on hardware sub-page
- Hook up delete functionality

Closes RCORE-684

# Test Plan

Unit tests have been added that cover the bulk of the page functionality. Storybook Stories have been created where appropriate, and verified visually. I also verified the page in both desktop mode and on the dev kit ODD.

# Changelog

- Used new tabbed buttons for navigation
- Added medium button hifi component
- Removed unnecessary medium rounded button component
- Improved linting on LargeButton component
- Added new typography for protocol header
- Set max number of pins as an app constant
- Generalized and styled Too Many Pins modal
- Added back chevron icon svg
- Applied Protocol details page hifi design
- Applied Hardware and Labware hifi styling

# Review requests

No special requests.

# Risk assessment

There is no risk to current users, as this work is confined to the ODD.

### CODE REVIEW

**Comment by ewagoner** [comment]
https://user-images.githubusercontent.com/2960/226750738-8127fc53-f06a-48ce-8649-1955cdd10dff.mov

File: app/src/App/constants.ts [hunk 1]
```diff
@@ -1,3 +1,6 @@
 // defines a constant for the nav bar width - used in run log component to calculate centering
 export const NAV_BAR_WIDTH = '5.625rem'
 export const SLEEP_NEVER_MS = 604800000
+
+// What is the maximum number of protocols one can pin? This many.
+export const MAXIMUM_PINNED_PROTOCOLS = 8
```
Comment (jerader) [inline app/src/App/constants.ts:6]: should this go in `shared-data/constants`? 🤔

File: app/src/assets/localization/en/protocol_info.json [hunk 2]
```diff
@@ -3,6 +3,7 @@
   "choose_file": "Choose File...",
   "choose_protocol_file": "Choose File",
   "choose_snippet_type": "Choose the Labware Offset Data Python Snippet based on target execution environment.",
+  "close": "Close",
```
Comment (jerader) [inline app/src/assets/localization/en/protocol_info.json]: you can use the `close` key from `shared.json` if you want!

File: app/src/atoms/buttons/OnDeviceDisplay/LargeButton.tsx [hunk 3]
```diff
@@ -118,7 +118,7 @@ export function LargeButton(props: LargeButtonProps): JSX.Element {
         name={iconName ?? 'play'}
         aria-label={`LargeButton_${iconName ?? 'play'}`}
         color={
-          disabled
+          disabled === true
```
Comment (jerader) [inline app/src/atoms/buttons/OnDeviceDisplay/LargeButton.tsx]: is this change required?

File: app/src/atoms/buttons/OnDeviceDisplay/MediumButton.tsx [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    MediumButtonTypes,
+    {
+      activeBackgroundColor: string
+      defaultBackgroundColor: string
+      defaultColor: string
+      iconColor: string
+    }
+  > = {
+    alert: {
+      activeBackgroundColor: '#b91f20',
```
Comment (jerader) [inline app/src/atoms/buttons/OnDeviceDisplay/MediumButton.tsx:59]: maybe add a `todo` comment here to remember to go back and add the constants.

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/Hardware.tsx [hunk 5]
```diff
@@ -27,22 +28,24 @@ const TableHeader = styled('th')`
...
-  height: 4rem;
+  height: 4.75rem;
 `
 
 const TableDatum = styled('td')`
+  font-size: ${TYPOGRAPHY.fontSize22};
+  font-weight: ${TYPOGRAPHY.lineHeight28};
   padding: ${SPACING.spacing2};
   white-space: break-spaces;
   text-overflow: wrap;
```
Comment (jerader) [inline app/src/pages/OnDeviceDisplay/ProtocolDetails/Hardware.tsx]:
```suggestion
  text-overflow: ${WRAP};
```

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx [hunk 6]
```diff
@@ -148,73 +221,101 @@ const ProtocolSectionContent = (
...
+        margin={SPACING.spacing4}
+      >
+        <MediumButton
+          buttonText={
+            pinned
+              ? t('protocol_info:unpin_protocol')
+              : t('protocol_info:pin_protocol')
+          }
+          buttonType="secondary"
+          iconName={'push-pin'}
```
Comment (jerader) [inline app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx]:
```suggestion
          iconName="push-pin"
```

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx [hunk 7]
```diff
@@ -148,73 +221,101 @@ const ProtocolSectionContent = (
...
+          onClick={handlePinClick}
+          width="30.375rem"
+        />
+        <MediumButton
+          buttonText={t('protocol_info:delete_protocol')}
+          buttonType="alertSecondary"
+          iconName={'delete'}
```
Comment (jerader) [inline app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx]:
```suggestion
          iconName="delete"
```

File: components/src/ui-style-constants/typography.ts [hunk 8]
```diff
@@ -17,11 +18,13 @@ export const fontSizeCaption = '0.625rem' // 10px
 
 // Font Weights
 export const fontWeightBold = 800
+export const fontWeightNearlyBold = 700
```
Comment (jerader) [inline components/src/ui-style-constants/typography.ts]: looks like figma names this `level2_bold`? so maybe `fontWeightLevel2_bold`? cc @koji for naming thoughts.

File: app/src/App/constants.ts (hunk 1 above)
Comment (koji) [inline app/src/App/constants.ts:6]: 👍

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/Hardware.tsx [hunk 9]
```diff
@@ -73,27 +76,51 @@ export const Hardware = (props: { protocolId: string }): JSX.Element => {
...
-              <TableDatum>{getHardwareLocation(hardware, t)}</TableDatum>
-              <TableDatum>{getHardwareName(hardware)}</TableDatum>
+            <TableRow key={id}>
+              <TableDatum>
+                <StyledText
+                  color={COLORS.darkBlack_hundred}
+                  lineHeight={TYPOGRAPHY.lineHeight28}
+                  paddingLeft={SPACING.spacing5}
+                >
+                  {getHardwareLocation(hardware, t)}
```
Comment (jerader) [inline app/src/pages/OnDeviceDisplay/ProtocolDetails/Hardware.tsx:119]: not sure if there is a separate ticket to address this but the UI for the hardware locations don't match what is in figma. Slots should only say the number with the border around them and mounts should be capitalized.

**Review by jerader — COMMENTED** [review]
I left a few clean up comments and questions. Looking good so far! Looks great on the simulator.

File: components/src/icons/icon-data.ts [hunk 10]
```diff
@@ -14,6 +14,11 @@ export const ICON_DATA_BY_NAME = {
     path:
       'M4,11V13H16L10.5,18.5L11.92,19.92L19.84,12L11.92,4.08L10.5,5.5L16,11H4Z',
   },
+  back: {
+    viewBox: '0 0 19 34',
+    path:
+      'M16.6667 33.6668L0 17.0002L16.6667 0.333496L19 2.7085L4.70833 17.0002L19 31.2918L16.6667 33.6668Z',
+  },
```
Comment (koji) [inline components/src/icons/icon-data.ts:21]: Question 
`chevron-left` doesn't match for the design ?

File: app/src/atoms/buttons/OnDeviceDisplay/LargeButton.tsx (hunk 3 above)
Comment (ewagoner) [reply app/src/atoms/buttons/OnDeviceDisplay/LargeButton.tsx]: It throws a linter warning without it. Not required, but it does get rid of the side-eye eslint was giving.

File: app/src/App/constants.ts (hunk 1 above)
Comment (ewagoner) [reply app/src/App/constants.ts:6]: I waffled about this. I chose here because it's an device-specific constant like the others in here, but it'd fit in the other place too.

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/Hardware.tsx (hunk 9 above)
Comment (koji) [inline app/src/pages/OnDeviceDisplay/ProtocolDetails/Hardware.tsx:119]: For mount, the design uses `Right Mount` so probably we would need to add `textTransform` to `StyledText`.

`textTransform={TYPOGRAPHY. textTransformCapitalize}`

Screenshot 2023-03-22 at 11 05 51 AM

File: components/src/icons/icon-data.ts (hunk 10 above)
Comment (ewagoner) [reply components/src/icons/icon-data.ts:21]: It does not. Shape and size is different enough that it can't be styled to match.

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/Hardware.tsx (hunk 9 above)
Comment (jgbowser) [reply app/src/pages/OnDeviceDisplay/ProtocolDetails/Hardware.tsx:119]: I'm not sure if there is a separate ticket or not, but this is the same thing I asked about a couple of weeks ago re: the Labware Setup screen. I was going to make a function to map from the old "slots 1-12" format to the "a1, a2, ... etc" format but I believe the decision was to hold off on that until we just get the expected format from the backend. As of right now both the module setup and labware setup screens are also displaying locations as "slot #"

Comment (ewagoner) [reply app/src/pages/OnDeviceDisplay/ProtocolDetails/Hardware.tsx:119]: It's not just that -- the design shows iconography and other sugar, so I left the contents of this table as a TODO item on this PR.

Comment (ewagoner) [reply app/src/pages/OnDeviceDisplay/ProtocolDetails/Hardware.tsx:119]: Not a separate ticket yet (that I know of) but it is one of the TODO items in the PR description that I plan on spinning off into new tickets when this gets merged.

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx [hunk 11]
```diff
@@ -98,25 +145,50 @@ const ProtocolSectionTabs = (props: ProtocolSectionTabsProps): JSX.Element => {
...
+      >
+        {props.description}
+      </StyledText>
+      <Flex
+        backgroundColor={COLORS.darkBlack_twenty}
+        fontSize={TYPOGRAPHY.fontSize22}
+        fontWeight={TYPOGRAPHY.fontWeightRegular}
+        lineHeight={TYPOGRAPHY.lineHeight28}
+        maxWidth="22rem"
+        padding={`${SPACING.spacing3} 0.75rem`}
```
Comment (koji) [inline app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx:195]:
```suggestion
        padding={`${SPACING.spacing3} 0.75rem`}
        borderRadius={BORDERS.size_one}
```

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx [hunk 12]
```diff
@@ -1,65 +1,110 @@
...
+        marginBottom={SPACING.spacing3}
+      >
+        <Btn onClick={() => history.goBack()}>
+          <Icon
+            name="back"
+            width="1.1875rem"
+            color={COLORS.darkBlack_hundred}
+          />
+        </Btn>
+        <Flex flexDirection={DIRECTION_COLUMN} maxWidth="42.625rem">
```
Comment (koji) [inline app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx]: Seems that we need to add space(8px) between chip and text from design.

```suggestion
        <Flex flexDirection={DIRECTION_COLUMN} maxWidth="42.625rem" gridGap={SPACING.spacing3}>
```

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx [hunk 13]
```diff
@@ -1,65 +1,110 @@
...
-        <StyledText as="h2">{`${t('protocol_info:date_added')}: ${
-          date != null
-            ? format(new Date(date), 'MM/dd/yyyy')
-            : t('shared:no_data')
-        }`}</StyledText>
-        <StyledText as="h2">{`${t(
-          'protocol_details:creation_method'
-        )}: ${getCreationMethod(protocolType)}`}</StyledText>
+      <Flex alignItems={ALIGN_CENTER} maxHeight="3.75rem" minWidth="15.6875rem">
+        <NewPrimaryBtn
```
Comment (koji) [inline app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx:106]: Can we use small button?

now the button link to small button.
Screenshot 2023-03-22 at 11 54 30 AM

File: app/src/App/constants.ts (hunk 1 above)
Comment (koji) [reply app/src/App/constants.ts:6]: @jerader Do we have any rules on constants?

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx (hunk 13 above)
Comment (ewagoner) [reply app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx:106]: Maybe. Yesterday, Figma called this a "custom CTA" so that's what I made. I'll see if I can use the common component instead.

File: components/src/ui-style-constants/typography.ts (hunk 8 above)
Comment (koji) [reply components/src/ui-style-constants/typography.ts]: oh I didn't notice that Mel added the name. Basically, using the name from Figma would be useful since probably we could avoid mis-communication that may caused by terms from the dev team and terms from the design team.

File: components/src/icons/icon-data.ts (hunk 10 above)
Comment (koji) [reply components/src/icons/icon-data.ts:21]: Ok, thanks!

File: app/src/atoms/buttons/OnDeviceDisplay/MediumButton.tsx (hunk 4 above)
Comment (koji) [reply app/src/atoms/buttons/OnDeviceDisplay/MediumButton.tsx:59]: @jerader Have you gotten any notifications about new color names from Mel?

File: app/src/atoms/buttons/OnDeviceDisplay/LargeButton.tsx (hunk 3 above)
Comment (koji) [reply app/src/atoms/buttons/OnDeviceDisplay/LargeButton.tsx]: I think right now we can use `disabled` for consistency. 
At some point, we will need to review the current our `eslint` configurations.

File: app/src/App/constants.ts (hunk 1 above)
Comment (jerader) [reply app/src/App/constants.ts:6]: I don't think there are rules, I've just been adding stuff to the `shared-data/constants` when I know someone or myself will use the constant elsewhere. But hmmm, maybe we should just keep it here since we won't be using this constant in another project like PD or anything.

File: app/src/atoms/buttons/OnDeviceDisplay/MediumButton.tsx (hunk 4 above)
Comment (jerader) [reply app/src/atoms/buttons/OnDeviceDisplay/MediumButton.tsx:59]: I have not yet, but I will ping her today to get the status

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/Hardware.tsx (hunk 9 above)
Comment (jerader) [reply app/src/pages/OnDeviceDisplay/ProtocolDetails/Hardware.tsx:119]: Got it, thanks yall! The change will need to be added to the Liquids setup screen as well.

File: app/src/atoms/buttons/OnDeviceDisplay/LargeButton.tsx (hunk 3 above)
Comment (ewagoner) [reply app/src/atoms/buttons/OnDeviceDisplay/LargeButton.tsx]: Ok. I'll change it back. Not a fan of ignoring warnings in my editor, especially when I'm a guest in someone else's code, but I understand your rules are in flux.

File: app/src/atoms/buttons/OnDeviceDisplay/MediumButton.tsx [hunk 14]
```diff
@@ -0,0 +1,170 @@
...
+import { StyledText } from '../../text'
+import type { IconName, StyleProps } from '@opentrons/components'
+
+type MediumButtonTypes =
+  | 'primary'
+  | 'secondary'
+  | 'alert'
+  | 'alertSecondary'
+  | 'tertiary'
+  | 'tertiaryLight'
```
Comment (koji) [inline app/src/atoms/buttons/OnDeviceDisplay/MediumButton.tsx]: Mel named `Tertiary High` so let's align as much as possible!

```suggestion
  | 'tertiaryHigh'
  | 'tertiaryHighLight'
```

File: app/src/atoms/buttons/OnDeviceDisplay/LargeButton.tsx (hunk 3 above)
Comment (koji) [reply app/src/atoms/buttons/OnDeviceDisplay/LargeButton.tsx]: Yeah, feel the same.

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/__tests__/ProtocolDetails.test.tsx [hunk 15]
```diff
@@ -0,0 +1,119 @@
...
+  })
+  it('renders the navigation buttons', () => {
+    const [{ getByRole }] = render()
+    getByRole('button', { name: 'Summary' })
+    getByRole('button', { name: 'Hardware' })
+    getByRole('button', { name: 'Labware' })
+    getByRole('button', { name: 'Liquids' })
+    getByRole('button', { name: 'Initial Deck Layout' })
+  })
+})
```
Comment (koji) [inline app/src/pages/OnDeviceDisplay/ProtocolDetails/__tests__/ProtocolDetails.test.tsx:140]: probably we would like to have cases that when tapping each button and call the each component.

```ts
const mocHardware = Hardware ....

beforeEach(() => {
  mockHardware.mockReturnValue(<div>Mock Hardware</div>)
})

it('render Hardware component when tapping Hardware button', () => {
  const [{ getByRole }] = render()
  const button = getByRole('button', { name: 'Hardware' })
  fireEvent.click(button)
  getByText('Mock Hardware')
}
```

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx (hunk 13 above)
Comment (koji) [reply app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx:106]: Jethary is addressing feedback from Mel so you can leave this now, but please leave the comment.

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx [hunk 16]
```diff
@@ -98,25 +145,50 @@ const ProtocolSectionTabs = (props: ProtocolSectionTabsProps): JSX.Element => {
...
+        )}: `}</StyledText>
         <StyledText>{props.author}</StyledText>
       </Flex>
-      <StyledText>{props.description}</StyledText>
+      <StyledText
+        fontSize={TYPOGRAPHY.fontSize22}
+        fontWeight={TYPOGRAPHY.fontWeightRegular}
+        lineHeight={TYPOGRAPHY.lineHeight28}
+      >
+        {props.description}
```
Comment (koji) [inline app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx]: Can we destruct props?
const { author, description, data } = props 
A few components do `props.something` but we avoid that.

```suggestion
        {description}
```

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx [hunk 17]
```diff
@@ -127,8 +199,9 @@ const ProtocolSectionContent = (
     case 'Summary':
       protocolSection = (
         <Summary
-          author={props.metadata.author ?? null}
-          description={props.metadata.description ?? null}
+          author={props.protocolData.metadata.author ?? null}
+          date={props.protocolData.createdAt ?? null}
+          description={props.protocolData.metadata.description ?? null}
```
Comment (koji) [inline app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx]: same as the above

```suggestion
          author={protocolData.metadata.author ?? null}
          date={protocolData.createdAt ?? null}
          description={protocolData.metadata.description ?? null}
```

File: app/src/pages/OnDeviceDisplay/TooManyPinsModal.tsx [hunk 18]
```diff
@@ -9,25 +9,18 @@ import {
...
 } from '@opentrons/components'
 
-import { StyledText } from '../../../atoms/text'
-import { ModalShell } from '../../../molecules/Modal'
-
-import type { UseLongPressResult } from '@opentrons/components'
+import { StyledText } from '../../atoms/text'
+import { ModalShell } from '../../molecules/Modal'
 
 export function TooManyPinsModal(props: {
```
Comment (koji) [inline app/src/pages/OnDeviceDisplay/TooManyPinsModal.tsx]: As I posted to Slack, Modal should be under `molecules` since we don't put `stories` under `pages` and `organisms`

Comment (koji) [reply app/src/pages/OnDeviceDisplay/TooManyPinsModal.tsx]: Let's create molecules/Modal/OnDeviceDisplay and move this into the new folder.

**Comment by koji** [comment]
TabbedButton focus still shows outline. Do you plan to update TabbedButton in a following PR?

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx (hunk 17 above)
Comment (ewagoner) [reply app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx]: Oh good catch. I meant to do this when I refactored this section.

**Comment by ewagoner** [comment]
> TabbedButton focus still shows outline. Do you plan to update TabbedButton in a following PR?

I saw that conversation in slack today. I'll fix both TabbedButton and MediumButton as a separate commit to this PR, since I'm here already using them both.

Edit: I didn't have the errant box shadow in the MediumButton, so I just fixed the TabbedButton.

**Comment by ewagoner** [comment]
I've addressed all the feedback on this PR. I do want to try to refactor the MediumButton to make it in line with this comment on the SmallButton PR: https://github.com/Opentrons/opentrons/pull/12340#pullrequestreview-1353402349

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx [hunk 19]
```diff
@@ -98,123 +150,180 @@ const ProtocolSectionTabs = (props: ProtocolSectionTabsProps): JSX.Element => {
...
+      <Flex
+        backgroundColor={COLORS.darkBlack_twenty}
+        borderRadius={BORDERS.size_one}
+        fontSize={TYPOGRAPHY.fontSize22}
+        fontWeight={TYPOGRAPHY.fontWeightRegular}
+        lineHeight={TYPOGRAPHY.lineHeight28}
+        maxWidth="22rem"
+        padding={`${SPACING.spacing3} 0.75rem`}
+      >
+        <StyledText>{`${t('protocol_info:date_added')}: ${
```
Comment (koji) [inline app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx:197]: last small thing need space between `description` and `date added`
in the design SPACING.spacing5 (24px)

Screenshot 2023-03-23 at 10 21 43 AM

There is gridGap(4px) so i guess the easiest way would be adding `marginTop: 20px` to L177 Flex

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx [hunk 20]
```diff
@@ -1,65 +1,114 @@
...
+          <StyledText
+            fontSize={TYPOGRAPHY.fontSize38}
+            fontWeight={TYPOGRAPHY.fontWeightLevel2_bold}
+            lineHeight={TYPOGRAPHY.lineHeight48}
+            onClick={toggleTruncate}
+          >
+            {displayedTitle}
+          </StyledText>
+        </Flex>
       </Flex>
```
Comment (koji) [inline app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx:99]: Seems that the space between `displayedTitle` and `Start setup` button is missing.

Screenshot 2023-03-23 at 10 57 01 AM

Comment (ewagoner) [reply app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx:99]: It looks like there is a minimum 40px gutter there now. There are possibly some other recent spacing adjustments in the design, so I'll go over that carefully.

Comment (koji) [reply app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx:99]: Okay, thank you!

**Comment by ewagoner** [comment]
I think this is finally good to go, again, for real this time. Re-requesting reviews!

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx [hunk 21]
```diff
@@ -1,65 +1,122 @@
...
+    >
+      <Flex
+        alignItems={ALIGN_CENTER}
+        gridGap={SPACING.spacing5}
+        marginBottom={SPACING.spacing3}
+      >
+        <Btn onClick={() => history.goBack()}>
+          <Icon
+            name="back"
+            width="1.1875rem"
```
Comment (koji) [inline app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx]: In Figma the size is 2.5rem(40px)

Screenshot 2023-03-24 at 3 06 46 PM

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx (hunk 19 above)
Comment (koji) [reply app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx:197]: still missing the space between description and date added.

Screenshot 2023-03-24 at 3 09 21 PM

Screenshot 2023-03-24 at 3 09 42 PM

**Review by koji — APPROVED** [review]
Leave a few comments but looks good to me.
@jerader got a dev kit so probably waiting her review would be good.

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx [hunk 22]
```diff
@@ -98,123 +158,182 @@ const ProtocolSectionTabs = (props: ProtocolSectionTabsProps): JSX.Element => {
...
-      <Flex margin={SPACING.spacing4}>
-        <Flex
-          cursor={'default'}
+      <Flex
+        flexDirection={DIRECTION_ROW}
+        gridGap={SPACING.spacing3}
+        justifyContent={JUSTIFY_SPACE_BETWEEN}
+        margin={SPACING.spacing4}
+      >
+        <MediumButton
```
Comment (koji) [inline app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx:318]: @ewagoner 

the padding is 40px because of medium button which is absolutely right but the medium button with an icon the padding is 24px.

Screenshot 2023-03-24 at 3 14 46 PM

Screenshot 2023-03-24 at 3 21 03 PM

Could you check about this to Mel when she back to work?

Comment (koji) [reply app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx:318]: I just noticed this.

https://www.figma.com/file/OIdG64Q5cgvEw82ish5Eee/Design-System-(Flex)?node-id=1401-60773&t=uQGHIngZbWBWPWTm-0

I guess handling 2cases with an icon and without icon would be fine but not sure that we really need to handling a couple of padding patterns.
 
cc: @shlokamin

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx (hunk 21 above)
Comment (ewagoner) [reply app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx]: Hmm. I had the button be 40px wide with the icon on the left, but that seems to have gotten dropped somewhere.

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx (hunk 22 above)
Comment (ewagoner) [reply app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx:318]: Fixed!

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx (hunk 21 above)
Comment (ewagoner) [reply app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx]: Fixed!

File: app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx (hunk 22 above)
Comment (ewagoner) [reply app/src/pages/OnDeviceDisplay/ProtocolDetails/index.tsx:318]: Fixed!
